boilerplate/app/components/screen/screen.tsx

```diff
@@ -32,6 +32,37 @@ function ScreenWithScrolling(props: ScreenProps) {
   const backgroundStyle = props.backgroundColor ? { backgroundColor: props.backgroundColor } : {}
   const insetStyle = { paddingTop: props.unsafe ? 0 : insets.top }
 
+  // The followings for <Screen preset='auto'/>
+  // This will automatically disables scrolling if content fits the screen.
+  const { height } = Dimensions.get('window')
+  const screenHeight = React.useRef(null)
+  const [scrollEnabled, setScrollEnabled] = React.useState(true)
+
+  const updateScrollState = () => {
+    if (props.preset === 'auto'){
+      // check whether if content fits the screen
+      const contentFitsScreen = screenHeight.current > height * presets.auto.offset.percent - presets.auto.offset.point
+        
+      // then toggle scroll state according to it, make sure it's not rendering twice
+      if (scrollEnabled && !contentFitsScreen) setScrollEnabled(false)
+      else if (!scrollEnabled && contentFitsScreen) setScrollEnabled(true)
```

Comment:
```suggestion
      // content is less than the size of the screen, so we can disable scrolling
      if (scrollEnabled && !contentFitsScreen) setScrollEnabled(false)
      
      // content is greater than the size of the screen, so let's enable scrolling
      if (!scrollEnabled && contentFitsScreen) setScrollEnabled(true)
```

Reply:
I'm fine with it. Any reason you removed that **else**? Much better to read for sure, but unnecessary computing.

Reply:
I went through it in [this stream](https://youtu.be/bvNZ5fSrAfE?t=2408) and honestly this one change took me many retries until I settled on this. Haha

Reply:
Oh, I've just watched it. You spend a lot of time on this one. Makes sense. No more headache on this.